catboost/private/libs/algo/greedy_tensor_search.cpp

```diff
@@ -1385,62 +1459,170 @@ static TNonSymmetricTreeStructure GreedyTensorSearchDepthwise(
         TVector<TIndexType> splittedLeafs;
         TVector<TIndexType> nextLevelLeafs;
         const size_t maxFeatureValueCount = CalcMaxFeatureValueCount(*fold, candidatesContexts);
-        for (TIndexType leafToSplit : curLevelLeafs) {
-            const auto& leafBounds = ctx->SampledDocs.LeavesBounds[leafToSplit];
+
+        const auto CalcBestScoreAndCandidate = [&data,&ctx,&scoreStDev,
+                                                &candidatesContexts, &fold,
+                                                &maxBucketCount,
+                                                &maxFeatureValueCount, &maxSplitEnsamples]
+                                                                      (TBucketStats* stats, TBucketStats* pstats,
+                                                                       TBucketStats* sstats, TIndexType id,
+                                                                       double& gainLocal,
+                                                                       const TCandidateInfo** bestSplitCandidateLocal,
+                                                                       TSplit& bestSplitLocal){
+            double bestScoreLocal = MINIMAL_SCORE; double scoreBeforeSplitLocal = 0;
```

Comment:
should be at most one declaration per line

consider moving closer to the first use

Reply:
done